packages/flutter/lib/src/services/undo_manager.dart

```diff
@@ -0,0 +1,129 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import '../../services.dart';
+
+/// The direction in which an undo action should be performed, whether undo or redo.
+enum UndoDirection {
+  /// Perform an undo action.
+  undo,
+
+  /// Perform a redo action.
+  redo
+}
+
+/// A low-level interface to the system's undo manager.
+///
+/// To receive events from the system undo manager, create an
+/// [UndoManagerClient] and set it as the [client] on [UndoManager].
+///
+/// The [setUndoState] method can be used to update the system's undo manager
+/// using the [canUndo] and [canRedo] parameters.
+///
+/// When the system undo or redo button is tapped, the current
+/// [UndoManagerClient] will receive [UndoManagerClient.handlePlatformUndo]
+/// with an [UndoDirection] representing whether the event is "undo" or "redo".
+///
+/// Currently, only iOS has an UndoManagerPlugin implemented on the engine side.
```

Comment:
Nit: Maybe add the following.  I'm looking at these docs and noticing that someone that lands on this page might not know what this class can actually be used for at a high level.

"On iOS, this can be used to listen to the keyboard undo/redo buttons and the undo/redo gestures."

Reply:
Someone else should double check me on this, but it seems to work on macOS without any further changes, I think because we're handling the UndoTextIntent and RedoTextIntent.

![Kapture 2022-11-08 at 11 59 53](https://user-images.githubusercontent.com/3249979/200640818-a9b3453a-1923-4d2d-9726-179e0c56175b.gif)